sdk/python/kfp/components/_components.py

```diff
@@ -230,8 +230,8 @@ def create_task_from_component_and_arguments(pythonic_arguments):
     import inspect
     from . import _dynamic
 
-    #Reordering the inputs since in Python optional parameters must come after reuired parameters
-    reordered_input_list = [input for input in inputs_list if not input.optional] + [input for input in inputs_list if input.optional]
+    #Reordering the inputs since in Python optional parameters must come after required parameters
+    reordered_input_list = [input for input in inputs_list if input.default is None and not input.optional] + [input for input in inputs_list if not (input.default is None and not input.optional)]
```

Comment:
nit: is it hard to add a unit test to cover this change?

Reply:
Thanks. Added two unit tests.